examples/solverdummies/cpp/solverdummy.cpp

```diff
@@ -10,23 +10,29 @@ int main(int argc, char **argv)
   using namespace precice;
   using namespace precice::constants;
 
-  if (argc != 4) {
+  std::string configFileName(argv[1]);
+  std::string solverName(argv[2]);
+  std::string meshName;
+
+  if (argc != 3) {
     std::cout << "Usage: ./solverdummy configFile solverName meshName\n\n";
     std::cout << "Parameter description\n";
     std::cout << "  configurationFile: Path and filename of preCICE configuration\n";
     std::cout << "  solverName:        SolverDummy participant name in preCICE configuration\n";
-    std::cout << "  meshName:          Mesh in preCICE configuration that carries read and write data\n";
     return 1;
   }
 
-  std::string configFileName(argv[1]);
-  std::string solverName(argv[2]);
-  std::string meshName(argv[3]);
-
-  std::cout << "DUMMY: Running solver dummy with preCICE config file \"" << configFileName << "\", participant name \"" << solverName << "\", and mesh name \"" << meshName << "\".\n";
+  std::cout << "DUMMY: Running solver dummy with preCICE config file \"" << configFileName << "\" and participant name \"" << solverName << "\".\n";
 
   SolverInterface interface(solverName, configFileName, commRank, commSize);
 
+  if (solverName == "SolverOne") {
+    meshName = "MeshOne";
+  }
+  if (solverName == "SolverTwo") {
+    meshName = "MeshTwo";
+  }
```

Comment:
```suggestion
  std::string dataWriteName;
  std::string dataReadName;
  if (solverName == "SolverOne") {
    meshName = "MeshOne";
    dataWriteName = "dataOne";
    dataReadName = "dataTwo";    
  }
  if (solverName == "SolverTwo") {
    meshName = "MeshTwo";
    dataReadName  = "dataOne";
    dataWriteName = "dataTwo";    
  }
```

This saves you the `if` below. Same for the C version of the solverdummy.